python/tests/test_eval.py

```diff
@@ -172,3 +174,138 @@ async def test_run_test_suite_with_inputs(data: st.DataObject):
             ).all()
         case Err(e):
             assert False, f"expected Ok, got Err({e})"
+
+
+def _extract_id(
+    metric_fn: SampleEvaluationFunction[T_OutputDatum], metric_input: T_OutputDatum
+) -> str:
+    return metric_fn(metric_input).interpretation.id
+
+
+def _extract_id_pair_for_input_pair(
+    metric: SampleEvaluationFunction[str], input_pair: tuple[str, str]
+) -> tuple[str, str]:
+    return (
+        _extract_id(metric_fn=metric, metric_input=input_pair[0]),
+        _extract_id(metric_fn=metric, metric_input=input_pair[1]),
+    )
+
+
+def _extract_id_pair_for_metric_pair(
+    metric_pair: tuple[SampleEvaluationFunction[str], SampleEvaluationFunction[str]],
+    metric_input: str,
+) -> tuple[str, str]:
```

Comment:
Can you give example comment docstrings on how input pair vs. metric pair is different?